go/libraries/doltcore/doltdb/table.go

```diff
@@ -110,6 +110,9 @@ func (t *Table) ValueReadWriter() types.ValueReadWriter {
 
 // NodeStore returns the NodeStore for this table.
 func (t *Table) NodeStore() tree.NodeStore {
+	if t == nil {
```

Comment:
Is this a principled use of a nil receiver? Does the nil `*Table` have a semantic meaning where we need to support this (as opposed to it just kicking the error along silently)?